lms/djangoapps/oauth_dispatch/tests/test_views.py

```diff
@@ -0,0 +1,142 @@
+"""
+Tests for Blocks Views
+"""
+
+import json
+
+import ddt
+from django.test import TestCase, RequestFactory
+from unittest import skip
+
+from student.tests.factories import UserFactory
+
+from .. import adapters
+from .. import views
+
+
+@ddt.ddt
+class TestAccessTokenView(TestCase):
+    """
+    Test class for AccessTokenView
+    """
+
+    dop_adapter = adapters.DOPAdapter()
+    dot_adapter = adapters.DOTAdapter()
+
+    def setUp(self):
+        super(TestAccessTokenView, self).setUp()
+        self.user = UserFactory()
+        self.dot_app = self.dot_adapter.create_public_client(user=self.user, client_id='dot-app-client-id')
+        self.dop_client = self.dop_adapter.create_public_client(user=self.user, client_id='dop-app-client-id')
+
+    def _post_access_token_request(self, user, client):
+        """
+        Make an HTTP POST request a token for the given user and client.
+
+        Returns an HTTP Response.
+        """
+        token_view = views.AccessTokenView.as_view()
+        request = RequestFactory().post('/', {
+            'client_id': client.client_id,
+            'grant_type': 'password',
+            'username': user.username,
+            'password': 'test',
+        })
+        return token_view(request)
+
+    @ddt.data('dop_client', 'dot_app')
+    def test_access_token_fields(self, client_attr):
+        client = getattr(self, client_attr)
+        response = self._post_access_token_request(self.user, client)
+        self.assertEqual(response.status_code, 200)
+        data = json.loads(response.content)
+        self.assertIn('access_token', data)
+        self.assertIn('expires_in', data)
+        self.assertIn('scope', data)
+        self.assertIn('token_type', data)
+
+    def test_dot_access_token_provides_refresh_token(self):
+        response = self._post_access_token_request(self.user, self.dot_app)
+        self.assertEqual(response.status_code, 200)
+        data = json.loads(response.content)
+        self.assertIn('refresh_token', data)
+
+    def test_dop_access_token(self):
+        response = self._post_access_token_request(self.user, self.dop_client)
+        self.assertEqual(response.status_code, 200)
+        data = json.loads(response.content)
+        self.assertNotIn('refresh_token', data)
```

Comment:
Note: DOP does not return `refresh_token` only for Public clients.  For private clients, it does.  So either add tests for private client, or update the name of the test method to specify that it's for public client only.